docs: simplify Terraform sections and convert to steps. (#29491)
gravitational/teleport: docs/pages/management/dynamic-resources/terraform-provider.mdx

---
title: Terraform Provider

> description: How to deploy infrastructure-as-code using the Teleport Terraform provider.

Reviewer: ```suggestion
description: How to manage dynamic resources as code using the Teleport Terraform provider.
```
For precision

- (!docs/pages/includes/tctl.mdx!)

## Step 1/3. Create Teleport credentials for Terraform

Terraform needs a signed identity file from the Teleport cluster certificate
authority to manage resources in the cluster.

> You can create a local Teleport user for this purpose or you can use a 

Reviewer: The product name is "Machine ID", not "a Machine ID". There are other instances of this as well.

Author: Personally, I'm not a big fan of this as a formal product name (generic terminology used as if it is a proper noun always strikes me as odd). I also see machine identity agent is sometimes used, so there still might be some inconsistency around the name.

[Machine ID](../../machine-id/introduction.mdx) to generate credentials.

If you intend to run Terraform from a CI/CD platform, a Machine ID is often a better 

> option for generating credentials. A Machine ID can provision ephemeral short-lived 

Reviewer 2: ```suggestion
option for generating credentials.  Machine ID can provision ephemeral short-lived 
```

Machine ID is one of our product lines, so I think this made more sense before. (We would say "Teleport can X", not "A Teleport can X")

   version: v6
   ---
   kind: user
   metadata:
     name: terraform
   spec:
     roles: ['terraform']
   version: v2
   ```
   
   These settings configure a user and role named `terraform` with the permissions 
   required to manage resources in your Teleport cluster.

1. Create the `terraform` user and role by running the following command:

   ```code
   $ tctl create terraform.yaml
   ```
   
   The `terraform` user can't sign in to get credentials, so you must have another user
   **impersonate** the `terraform` account to request a certificate.

1. Create a new file called `terraform-impersonator.yaml` and open it in an editor.

1. Configure a role that enables your user to impersonate the Terraform user by pasting 
the following content into the `terraform-impersonator.yaml` file:
   
   ```yaml
   kind: role
   version: v6
   metadata:
     name: terraform-impersonator
   spec:
     allow:
       # This impersonate rule will allow any user with this role to impersonate
       # and generate certificates for the user named "terraform" with a role also
       # named "terraform".
       impersonate:
         users: ['terraform']
         roles: ['terraform']
   ```

> 1. Create the terraform-impersonator role by running the following command:

Reviewer: ```suggestion
1. Create the `terraform-impersonator` role by running the following command:
```
